host/src/attribute.rs

```diff
@@ -393,12 +395,12 @@ impl<'d, M: RawMutex, const MAX: usize> AttributeTable<'d, M, MAX> {
     /// The return value of the closure is returned in this function and is assumed to be infallible.
     ///
     /// If the characteristic for the handle cannot be found, an error is returned.
-    pub fn get<F: FnMut(&[u8]) -> T, T>(&self, handle: Characteristic, mut f: F) -> Result<T, Error> {
+    pub fn get<T: GattValue>(&self, handle: &Characteristic<T>) -> Result<T, Error> {
```

Comment:
I think `get` and `set` should operate on Attributes generally, not just Characteristics? You also need to be able to get and set descriptor values and the like.

Reply:
Oh I didn't realise this! Ok that might need some further thought and development then.

Reply:
I think you're probably right about this, but I think that as this is the way the attribute table was working before it's worth tackling in a separate PR to avoid this one becoming too large. I'll raise an issue for it.

Reply:
Raised #155 to implement this.

Reply:
Agreed, it can be in a separate PR.